File: src/main/java/nieboczek/lifestolen/serializer/base/FloatSerializer.java
package nieboczek.lifestolen.serializer.base;

import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder;
import nieboczek.lifestolen.serializer.lang.TokenStream;
import nieboczek.lifestolen.serializer.lang.TokenType;

public class FloatSerializer extends Serializer<Float> {
    private FloatSerializer() {
    }

    public static FloatSerializer of() {
        return new FloatSerializer();
    }

    @Override
    public void serialize(Float value, SerializedStringBuilder builder) {
        builder.text(value.toString());
    }

    @Override
    public Float deserialize(TokenStream stream) {
        return Float.parseFloat(stream.nextTokenText(TokenType.NUMBER));
    }
}

File: src/main/kotlin/nieboczek/lifestolen/serializer/base/FloatSerializer.kt
package nieboczek.lifestolen.serializer.base

import nieboczek.lifestolen.serializer.SerializerError
import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder
import nieboczek.lifestolen.serializer.lang.TokenStream
import nieboczek.lifestolen.serializer.lang.TokenType

class FloatSerializer : Serializer<Float>() {
    override fun serialize(value: Float, builder: SerializedStringBuilder) {
        builder.text(value.toString())
    }

    override fun deserialize(stream: TokenStream): Float {
        val text = stream.nextTokenText(TokenType.NUMBER)
        try {
            return text.toFloat()
        } catch (e: NumberFormatException) {
            throw SerializerError("Invalid float value: \"$text\"", e)
        }
    }
}
